pbcore/io/align/_BamSupport.py

```diff
@@ -67,7 +68,8 @@ def reverseComplementAscii(a):
 # qId calculation from RG ID string
 #
 def rgAsInt(rgIdString):
-    return np.int32(int(rgIdString.split("/")[0], 16))
+    return np.int32(int(re.sub("-", "", rgIdString.split("/")[0]), 16)
+                    % (np.iinfo(np.int32).max+1))
```

Comment:
I don't think the modulus will give the intended result, which in the case of overflow will be a negative number.  Based on my quick experimenting with one of the failing unit tests, if it overflows I think you need to take the modulus and then subtract `np.iinfo(np.int32).max+1` again.

Reply:
I had to scratch my head, but I think I see your point.  Legacy behavior is: the overflow wraps around at np.iinfo(np.int32).min.  Meanwhile my implementation is: wrap around at 0 straight.  To be fully legacy compatible, my modulus wraparound is incorrect and I need to compensate so the index wraps between np.iinfo(np.int32).min and np.iinfo(np.int32).max.  Do I understand correctly?

Reply:
I have pushed a commit, this may clear things.